services/horizon/internal/db2/core/transaction.go

```diff
@@ -35,10 +42,36 @@ func (tx *Transaction) EnvelopeXDR() string {
 }
 
 // Fee returns the fee that was paid for `tx`
-func (tx *Transaction) Fee() int32 {
+func (tx *Transaction) Fee(whiteListData map[string]string) int32 {
+	if tx.isTxWhitelisted(whiteListData) {
+		return int32(0)
+	}
 	return int32(tx.Envelope.Tx.Fee)
 }
 
+func (tx *Transaction) isTxWhitelisted(whiteListData map[string]string) bool {
+	// First check if the source account is a Whitelist account.
+	if _, found := whiteListData[tx.Envelope.Tx.SourceAccount.Address()]; found {
+		return true
```

Comment:
This is not sufficient to prove it was whitelisted, even though it will be correct more often than not.

There are signature types other than ED25119 that may be used, which means that the source account may not have actually signed the tx.  This might be considered an oversight by the whitelist code in core (i.e. if the source is whitelisted, we shouldn't care about the signature), but at present this is the case.

Reply:
Solved, added TODO and Jira ticket